### PR TITLE
Remove the static constructors exposing autoRefresh

### PR DESCRIPTION
**Pull request by emanuelez**
Also adds static constructors based on File instead of Context

### CODE REVIEW

**Comment by emanuelez** [comment]
Changelog:
 * Remove deprecated static constructors
 * Introduce new static constructors based on File instead of Context, allowing to save Realm files in custom locations

**Comment by emanuelez** [comment]
@bmunkholm @cmelchior @kneth

**Comment by kneth** [comment]
You have to do want you promise: deprecated stuff must be removed. :+1:

File: realm/src/main/java/io/realm/Realm.java [hunk 1]
```diff
@@ -309,59 +282,49 @@ public static Realm getInstance(Context context, boolean autoRefresh) {
...
-     *                    should be automatically refreshed with the event loop (requires to be in a thread with a Looper)
+     * @param context an Android {@link android.content.Context}
+     * @param key     a 64-byte encryption key
      * @return an instance of the Realm class
      * @throws RealmMigrationNeededException The model classes have been changed and the Realm
      *                                       must be migrated
      * @throws RealmIOException              Error when accessing underlying file
-     * @throws java.lang.IllegalStateException The Realm is being instantiated with auto-refresh
-     *                                         in a Thread without a {@link android.os.Looper}
+     * @throws java.lang.IllegalStateException The Realm is being instantiated in a Thread without
```
Comment (cmelchior) [inline realm/src/main/java/io/realm/Realm.java]: Isn't that wrong now. It only happens if you manually call setAutorefresh, right?

File: realm/src/main/java/io/realm/Realm.java [hunk 2]
```diff
@@ -309,59 +282,49 @@ public static Realm getInstance(Context context, boolean autoRefresh) {
...
      * @param key     a 64-byte encryption key
-     * @param autoRefresh whether the Realm object and its derived objects (RealmResults and RealmObjects)
-     *                    should be automatically refreshed with the event loop (requires to be in a thread with a Looper)
      * @return an instance of the Realm class
      * @throws RealmMigrationNeededException The model classes have been changed and the Realm
      *                                       must be migrated
      * @throws RealmIOException              Error when accessing underlying file
-     * @throws java.lang.IllegalStateException The Realm is being instantiated with auto-refresh
-     *                                         in a Thread without a {@link android.os.Looper}
+     * @throws java.lang.IllegalStateException The Realm is being instantiated in a Thread without
```
Comment (cmelchior) [inline realm/src/main/java/io/realm/Realm.java]: Same...

**Comment by cmelchior** [comment]
Apart from comment. Yay less code :+1:

**Pull request merged**